src/Umbraco.Core/Routing/DomainUtilities.cs

```diff
@@ -313,14 +315,10 @@ private static DomainAndUri GetByCulture(IReadOnlyCollection<DomainAndUri> domai
         /// <param name="domains">The domains.</param>
         /// <param name="uri">The uri, or null.</param>
         /// <returns>The domains and their normalized uris, that match the specified uri.</returns>
-        internal static IEnumerable<DomainAndUri> SelectDomains(IEnumerable<Domain> domains, Uri uri)
-        {
+        [return: NotNullIfNotNull(nameof(domains))]
```

Comment:
There's no need to create new objects if it's null, as the `SelectDomain` method does `if (domainsAndUris is null || domainsAndUris.Length == 0) return null;` anyway... And because `DomainsForNode` already ensured `domains` is not null, adding this attribute will ensure it knows the return value will not be null in that case as well, optimizing both calls 😄